zapcore/encoder.go

```diff
@@ -196,6 +197,31 @@ func (e *CallerEncoder) UnmarshalText(text []byte) error {
 	return nil
 }
 
+// A LoggerNameEncoder serializes a LoggerName to a primitive type.
+type LoggerNameEncoder func(string, PrimitiveArrayEncoder)
+
+// FullLoggerNameEncoder serializes a logger name as is
+func FullLoggerNameEncoder(loggerName string, enc PrimitiveArrayEncoder) {
+	enc.AppendString(loggerName)
+}
+
+// CapitalLoggerNameEncoder serializes a logger name in all caps
+func CapitalLoggerNameEncoder(loggerName string, enc PrimitiveArrayEncoder) {
+	enc.AppendString(strings.ToUpper(loggerName))
+}
```

Comment:
Rather than exporting this, let's put it into the test code. That'll give unit tests a way to exercise this functionality without adding this unusual option to the public API.